### ISSUE TITLE
Getting "generic name" of the dependency instead of the artifactID

### ISSUE THREAD

**Issue by Ali0600** [spdx/spdx-maven-plugin#136]
The issue I'm having is that with the SPDX files that are being generated a lot of the dependencies are putting the a "generic name" of the dependency instead of the artifactID.

For example.
        {
            "SPDXID": "SPDXRef-gnrtd140",
            **"name": "Apache XML Graphics Commons",**
            "downloadLocation": "NOASSERTION",
            "filesAnalyzed": false,
            "licenseConcluded": "NOASSERTION",
            "licenseDeclared": "Apache-2.0",
            "copyrightText": "UNSPECIFIED",
            "versionInfo": "2.4",
            "summary": "Apache XML Graphics Commons is a library that consists of several reusable \n    components used by Apache Batik and Apache FOP. Many of these components \n    can easily be used separately outside the domains of SVG and XSL-FO.",
            "originator": "Organization: Apache Software Foundation",
            "description": "Apache XML Graphics Commons is a library that consists of several reusable \n    components used by Apache Batik and Apache FOP. Many of these components \n    can easily be used separately outside the domains of SVG and XSL-FO.",
            "homepage": "http://xmlgraphics.apache.org/commons/"
        },

Is there a way we can get it to put the artifactID in the name instead of a "generic name", similar to this one. Or is there a configuration that can do this?
        {
            "SPDXID": "SPDXRef-gnrtd135",
            **"name": "org.apache.xmlgraphics:batik-svg-dom"**,
            "downloadLocation": "NOASSERTION",
            "filesAnalyzed": false,
            "licenseConcluded": "NOASSERTION",
            "licenseDeclared": "Apache-2.0",
            "copyrightText": "UNSPECIFIED",
            "versionInfo": "1.12",
            "summary": "Batik SVG DOM implementation",
            "originator": "Organization: Apache Software Foundation",
            "description": "Batik SVG DOM implementation",
            "homepage": "http://xmlgraphics.apache.org/batik/batik-svg-dom/"
        },

**Comment by hboutemy** [comment]
SPDX `name` is currently using pom.xml `project.name` value: https://maven.apache.org/ref/3.9.5/maven-model/maven.html#class_project

@goneall it's easy to implmeent SPDX `name` as `${project.groupId}:${artifactId}` = what @Ali0600 suggests
the only question is to take the decision

**Comment by goneall** [comment]
In looking at the code, it looks like we already use the `artifactId` if the project name is not specified.  We could change the default to include the groupId.  Note that the package name is set for the main package in `CreateSpdxMojo.getSpdxProjectInfoFromParameters( SpdxDocumentBuilder builder )` while the name is set for dependencies in `SpdxDependencyInformation.createSpdxPackage( ... )`

I personally would like to preserve the project name specified in the Maven POM file in the SPDX document, but I don't feel strongly about it.

One way to capture the full artifact and group ID is to create [package URL](https://github.com/package-url/purl-spec) [external reference](https://spdx.github.io/spdx-spec/v2.3/package-information/#721-external-reference-field).  The format would be `pkg:maven/${project.groupId}/${project.artifactId}@${project.version}`. - reference [purl spec for Maven](https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#maven).

@Ali0600 - let me know if adding that external reference information to the output would solve the issue.

**Comment by Ali0600** [comment]
@goneall Can we have the ability to choose through a configuration option or command line parameter? Using ${project.groupId}:${artifactId} instead of the default. The reason this would be useful is because there are tools out there that try to map the entries to the vulnerability database and these tools are not able to map on friendly name fields.

**Comment by goneall** [comment]
> @goneall Can we have the ability to choose through a configuration option or command line parameter? Using ${project.groupId}:${artifactId} instead of the default.

@Ali0600 I'm not opposed to a configuration option.  Let me know if you have a specific approach / parameter in mind.

> The reason this would be useful is because there are tools out there that try to map the entries to vulnerability databases and these tools are not able to map on friendly name fields.

It is better to use external references like the package URL for correlating with vulnerability databases since they have a well defined syntax.  If we generate purls, that may solve the problem.

**Comment by Ali0600** [comment]
@goneall Would "useArtifactID" as the configuration name be ok? With just setting it to "true" to apply it.

**Comment by goneall** [comment]
> @goneall Would "useArtifactID" as the configuration name be ok? With just setting it to "true" to apply it.

Seems reasonable.  @hboutemy - what do you think?  If you agree, I can create a PR with the change.

**Comment by Ali0600** [comment]
Hello, don't mean to rush anyone but any updates? @goneall @hboutemy

**Comment by Ali0600** [comment]
Hello @goneall @hboutemy , any updates on whether this can be implemented? Thank you.

**Comment by goneall** [comment]
@Ali0600 - PR #139 adds a configuration parameter `useArtifactID`:

```
    /**
     * If true, use ${project.groupId}:${artifactId} as the SPDX package name.
     * Otherwise, ${project.name} will be used
     */
    @Parameter
    private boolean useArtifactID;
```

If this approach works for you, I'll merge the PR.

**Comment by Ali0600** [comment]
Yes this works!

**Comment by goneall** [comment]
@Ali0600 - the code is now merged.  Let me know if need a new release of the plugin with this fix.

**Comment by Ali0600** [comment]
@goneall Awesome, yes I would need a new release with the fix so I can implement it. Thank you!

**Comment by Ali0600** [comment]
@goneall Hello, I see that the websites mentions the latest version 0.7.1-snapshot, but it isn't on maven central or I can't find it anywhere on the website, would it be possible to get it on maven central? Thank you.

**Comment by goneall** [comment]
> @goneall Hello, I see that the websites mentions the latest version 0.7.1-snapshot, but it isn't on maven central or I can't find it anywhere on the website, would it be possible to get it on maven central? Thank you.

I don't believe that Maven will allow for SNAPSHOT releases.  I'm just about to release a new version of the SPDX libraries - once that is done I'll do a new release of the plugin.

**Comment by goneall** [comment]
@Ali0600 I just published a new release.  It will probably take a while before it shows up on Maven central.

**Comment by Ali0600** [comment]
Hello @goneall , thank you for releasing a new version. I used the newest version, 0.7.1, and I used the useArtifactID parameter, but it doesn't seem to work. I used "true" and "True", but it didn't work.
This is how I setup the parameter.
    	<plugin>
          <groupId>org.spdx</groupId>
          <artifactId>spdx-maven-plugin</artifactId>
          
          <version>0.7.1</version>
          <executions>
              <execution>
                  <id>build-spdx</id>
                  <goals>
                      <goal>createSPDX</goal>
                  </goals>
 		   <phase>verify</phase>
              </execution>
          </executions>
          <configuration>
	    **<useArtifactID>true</useArtifactID>**
            
          </configuration>
    	</plugin>

The command I'm using is just "mvn clean verify". Is there maybe a --version flag I have to add? Or am I missing a configuration possibly?

**Comment by goneall** [comment]
I'll take a look when I get back to my laptop

**Comment by goneall** [comment]
@Ali0600 - I just tested on my local machine and it seems to work.

There's the output SPDX name: `"name" : "org.spdx:java-spdx-library",`

Here's the input configuration:

```
			<plugin>
				<groupId>org.spdx</groupId>
				<artifactId>spdx-maven-plugin</artifactId>
				<version>0.7.1</version>
				<executions>
					<execution>
						<id>build-spdx</id>
						<phase>verify</phase>
						<goals>
							<goal>createSPDX</goal>
						</goals>
					</execution>
				</executions>
				<configuration>
				    <useArtifactID>true</useArtifactID>
...
```

Note that if you are running just the verify command, the spdx file will show up in the target directory, not in the repository.  You need to run `mvn install` to copy the SPDX file into the repo.

**Comment by Ali0600** [comment]
Hi @goneall, I copied the exact input configuration you put and I'm still not getting the desired output. For all the spdx files, it is only the first dependency that shows the name output as such. 
{
  "SPDXID" : "SPDXRef-DOCUMENT",
  "spdxVersion" : "SPDX-2.3",
  "creationInfo" : {
    "created" : "2023-12-19T06:38:32Z",
    "creators" : [ "Tool: spdx-maven-plugin" ],
    "licenseListVersion" : "3.22"
  },
  **"name" : "com.test:reports",**
  "dataLicense" : "CC0-1.0",
  "documentNamespace" : "http://spdx.org/spdxpackages/com.test_reports-2.00.0",
  "packages" : [ {
    "SPDXID" : "SPDXRef-gnrtd0",
    "checksums" : [ {
      "algorithm" : "SHA1",
      "checksumValue" : "e85ca799b4e205d242f59feffc0a5d63cdf5a58d"
    } ],
    "copyrightText" : "NOASSERTION",
    "description" : "Spring Boot Dependencies",
    "downloadLocation" : "NOASSERTION",
    "filesAnalyzed" : true,
    "homepage" : "https://spring.io/projects/spring-boot/masterpom/reports",
    "licenseConcluded" : "Apache-2.0",
    "licenseDeclared" : "Apache-2.0",
    "licenseInfoFromFiles" : [ "NOASSERTION" ],
    "name" : "com.test:reports",
    "packageFileName" : "reports-2.00.0.jar",
    "packageVerificationCode" : {
      "packageVerificationCodeValue" : "d40067030d313631966d8a0947f0f4a6be04222b"
    },

However, for the rest, here is how they appear.

  }, {
    "SPDXID" : "SPDXRef-gnrtd80",
    "copyrightText" : "UNSPECIFIED",
    "description" : "Hibernate's core ORM functionality",
    "downloadLocation" : "NOASSERTION",
    "filesAnalyzed" : false,
    "homepage" : "http://hibernate.org/orm",
    "licenseConcluded" : "NOASSERTION",
    "licenseDeclared" : "NOASSERTION",
    **"name" : "Hibernate ORM - hibernate-core",**
    "originator" : "Organization:Hibernate.org",
    "summary" : "Hibernate's core ORM functionality",
    "versionInfo" : "5.4.32.Final"
  }, {
    "SPDXID" : "SPDXRef-gnrtd81",
    "copyrightText" : "UNSPECIFIED",
    "description" : "Javassist (JAVA programming ASSISTant) makes Java bytecode manipulation\n    simple.  It is a class library for editing bytecodes in Java.",
    "downloadLocation" : "NOASSERTION",
    "filesAnalyzed" : false,
    "homepage" : "http://www.javassist.org/",
    "licenseConcluded" : "NOASSERTION",
    "licenseDeclared" : "MPL-1.1",
    **"name" : "Javassist",**
    "originator" : "Organization:Shigeru Chiba, www.javassist.org",
    "summary" : "Javassist (JAVA programming ASSISTant) makes Java bytecode manipulation\n    simple.  It is a class library for editing bytecodes in Java.",
    "versionInfo" : "3.27.0-GA"
  }, {

Can you please check if the bottom dependencies are showing the artifactID.

**Comment by goneall** [comment]
@Ali0600 - I'm now able to reproduce the issue.  There must be a code path that is not picking up the configuration parameter for the package name.  Sorry about missing it the first time around.  I'll open a separate issue to track.
